net/node/infoUpdate.go

```diff
@@ -152,8 +151,8 @@ func (node *node) TryConnect() {
 	for _, n := range node.nbrNodes.List {
 		if n.GetState() == INACTIVITY && n.tryTimes < 3 {
 			//try to connect
-			node.reconnect(n)
 			n.tryTimes++
+			node.reconnect(n)
```

Comment:
If connected tryTimes should be 0, otherwise it should be return before set tryTimes be 0